frame/dex/src/lib.rs

```diff
@@ -679,29 +682,64 @@ pub mod pallet {
 		}
 
 		/// Used by the RPC service to provide current prices.
-		pub fn quote_price(
-			asset1: Option<u32>,
-			asset2: Option<u32>,
-			amount: u64,
+		pub fn quote_price_exact_tokens_for_tokens(
+			asset1: Option<T::AssetId>,
+			asset2: Option<T::AssetId>,
+			amount: AssetBalanceOf<T>,
+			include_fee: bool,
 		) -> Option<AssetBalanceOf<T>> {
 			let into_multi = |asset| {
 				if let Some(asset) = asset {
-					MultiAssetId::Asset(T::AssetId::from(asset))
+					MultiAssetId::Asset(asset)
 				} else {
 					MultiAssetId::Native
 				}
 			};
 			let asset1 = into_multi(asset1);
 			let asset2 = into_multi(asset2);
-			let amount = amount.into();
 			let pool_id = Self::get_pool_id(asset1, asset2);
 			let pool_account = Self::get_pool_account(pool_id);
-			let (pool_asset1, _) = pool_id;
 
-			let balance1 = Self::get_balance(&pool_account, asset1);
-			let balance2 = Self::get_balance(&pool_account, asset2);
+			let balance1 = Self::get_balance(&pool_account, asset1).ok()?;
+			let balance2 = Self::get_balance(&pool_account, asset2).ok()?;
 			if !balance1.is_zero() {
-				Self::quote(&amount, &balance1, &balance2).ok()
+				if include_fee {
+					Self::get_amount_out(&amount, &balance1, &balance2).ok()
+				} else {
+					Self::quote(&amount, &balance1, &balance2).ok()
+				}
+			} else {
+				None
+			}
+		}
+
+		/// Used by the RPC service to provide current prices.
+		pub fn quote_price_tokens_for_exact_tokens(
```

Comment:
I don't see any references to that function

Reply:
have updated decl_runtime_apis to reference it.